Amazon.IonObjectMapper/IonSerializer.cs

```diff
@@ -504,5 +542,33 @@ private IIonSerializer GetPrimitiveSerializer<T>(IIonSerializer defaultSerialize
 
             return defaultSerializer;
         }
+
+        private IIonSerializer CreateCustomSerializer (object item)
+        {
+            var customSerializerAttribute = item.GetType().GetCustomAttribute<IonSerializerAttribute>();
+            if (customSerializerAttribute.Factory != null) {
+                var customSerializerFactory = (IIonSerializerFactory)Activator.CreateInstance(customSerializerAttribute.Factory);
+                var customSerializer = customSerializerFactory.Create(options, options.CustomContext);
+                return customSerializer;
+            } else if (customSerializerAttribute.Serializer != null) {
+                // Create customSerializer with serializer attribute
+            } 
+
+            return null;
+        }
+
+        private IIonSerializer CreateCustomSerializerForDeserializer (Type type)

```

Comment:
This code is duplicated, no? Why do we need this method?

Reply:
Pass in value types are different between serializer and deserializer.  Fixed this by changing the pass in value type to the same one.